src/main/java/com/shapesecurity/shift/es2016/semantics/Explicator.java

```diff
@@ -796,8 +776,10 @@ NodeWithValue explicateExpression(Expression expression, boolean keepValue, bool
 				this.currentState.exitInlineFunction();
 				return functionBlock;
 			}
-			// abort on direct eval. todo maybe warn.
 			if (c.callee instanceof IdentifierExpression && (((IdentifierExpression) c.callee).name.equals("eval"))) {
+				if (this.throwOnEval) {
```

Comment:
How about instead we have them pass in a function that returns the list of instructions to use for a direct eval? Default to the list containing just `Halt`, but this also allows them to throw a `RuntimeException`.

Reply:
Done.